espresso-server/app/src/androidTest/java/io/appium/espressoserver/EspressoServerRunnerTest.java

```diff
@@ -37,14 +37,16 @@
 @RunWith(AndroidJUnit4.class)
 @LargeTest
 public class EspressoServerRunnerTest {
+    private Server espressoServer = Server.getInstance();
 
     @Test
     public void startEspressoServer() throws InterruptedException, IOException, DuplicateRouteException {
-        new Server();
-
-        // The server should run permanently. The test gets closed externally.
-        Thread.sleep(Long.MAX_VALUE);
+        espressoServer.start();
 
+        while (!espressoServer.isStopRequestReceived()) {
```

Comment:
I'd rather add a reasonable timeout here

Reply:
Hmm, not really sure what timeout value would make sense here 🤔.

Reply:
I think it might be useful to put a comment there, like `// keep serving until the stop request is received`